src/java.base/share/classes/jdk/internal/ref/PhantomCleanable.java

```diff
@@ -83,6 +92,9 @@ public PhantomCleanable(T referent, Cleaner cleaner) {
     /**
      * Insert this PhantomCleanable after the list head.
      */
+    /*[IF CRIU_SUPPORT]*/
+    @NotCheckpointSafe
+    /*[ENDIF] CRIU_SUPPORT */
```

Comment:
Note:
`PhantomCleanable.insert()` was the method that caused `JVMRestoreException` in all occurrences.
Adding `@NotCheckpointSafe` to the other two synchronization methods on `list` is fine as well.